specification/compute/Gallery.Management/models.tsp

```diff
@@ -1786,6 +1827,195 @@ model GallerySoftDeletedResourceProperties {
   softDeletedTime?: string;
 }
 
+/**
+ * Describes the properties of a gallery script definition.
+ */
+@added(Versions.v2025_03_03)
+model GalleryScriptProperties {
+  /**
+   * The description of this gallery script definition resource. This property is updatable.
+   */
+  description?: string;
+
+  /**
+   * The Eula agreement (End User License Agreement) for the gallery Script Definition.
+   */
+  eula?: string;
+
+  /**
+   * The privacy statement uri.
+   */
+  privacyStatementUri?: string;
+
+  /**
+   * The release note uri.
+   */
+  releaseNoteUri?: string;
+
+  /**
+   * The end of life date of the gallery Script Definition. This property can be used for decommissioning purposes. This property is updatable.
+   */
+  endOfLifeDate?: utcDateTime;
+
+  /**
+   * This property allows you to specify the supported type of the OS that application is built for. Possible values are: **Windows,** **Linux.**
+   */
+  supportedOSType: OperatingSystemTypes;
+
+  /**
+   * The provisioning state, which only appears in the response.
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: GalleryProvisioningState;
+}
+
+/**
+ * Specifies information about the gallery Script Definition that you want to update.
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "For backward compatibility"
+@added(Versions.v2025_03_03)
+model GalleryScriptUpdate extends UpdateResourceDefinition {
+  /**
+   * Describes the properties of a gallery script definition.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-private-usage" "For backward compatibility"
+  @Azure.ResourceManager.Private.conditionalClientFlatten
+  properties?: GalleryScriptProperties;
```

Comment:
@viveklingaiah 
I am getting this error: 
<img width="1370" height="137" alt="image" src="https://github.com/user-attachments/assets/dba2ffa2-3984-4238-b6fc-0c39c8069b46" />
https://github.com/Azure/azure-rest-api-specs/actions/runs/16300595655

should we create a model called "GalleryScriptUpdateProperties" that does not have provisioningState?

Reply:
No. provisioningState should be a readonly property. It is returned in response. User won't change / should not be able to change it while sending request